pkg/api/handlers/mcp.go

```diff
@@ -1577,3 +1578,124 @@ func (m *MCPHandler) ClearOAuthCredentials(req api.Context) error {
 	req.WriteHeader(http.StatusNoContent)
 	return nil
 }
+
+func (m *MCPHandler) GetServerDetails(req api.Context) error {
+	if !m.mcpSessionManager.KubernetesEnabled() {
+		return types.NewErrNotFound("Kubernetes is not enabled")
+	}
+
+	_, serverConfig, err := ServerForAction(req)
+	if err != nil {
+		return err
+	}
+
+	details, err := m.mcpSessionManager.GetServerDetails(req.Context(), serverConfig)
+	if err != nil {
+		return err
+	}
+
+	return req.Write(details)
+}
+
+func (m *MCPHandler) StreamServerLogs(req api.Context) error {
+	if !m.mcpSessionManager.KubernetesEnabled() {
+		return types.NewErrNotFound("Kubernetes is not enabled")
+	}
+
+	_, serverConfig, err := ServerForAction(req)
+	if err != nil {
+		return err
+	}
+
+	logs, err := m.mcpSessionManager.StreamServerLogs(req.Context(), serverConfig)
+	if err != nil {
+		return err
+	}
+
+	defer logs.Close()
+
+	// Set up Server-Sent Events headers
+	req.ResponseWriter.Header().Set("Content-Type", "text/event-stream")
+	req.ResponseWriter.Header().Set("Cache-Control", "no-cache")
+	req.ResponseWriter.Header().Set("Connection", "keep-alive")
+	req.ResponseWriter.Header().Set("Access-Control-Allow-Origin", "*")
+	req.ResponseWriter.Header().Set("Access-Control-Allow-Headers", "Cache-Control")
+
+	flusher, ok := req.ResponseWriter.(http.Flusher)
+	if !ok {
+		return fmt.Errorf("response writer does not support flushing")
+	}
+
+	// Send initial connection event
+	fmt.Fprintf(req.ResponseWriter, "event: connected\ndata: Log stream started\n\n")
+	flusher.Flush()
+
+	// Channel to coordinate between goroutines
+	logChan := make(chan string, 100) // Buffered to prevent blocking
+
+	// Start a goroutine to read logs
+	go func() {
+		defer close(logChan)
+
+		buf := make([]byte, 1024)
+		for {
+			select {
+			case <-req.Context().Done():
+				return
+			default:
+				n, err := logs.Read(buf)
+				if n > 0 {
+					// Split by lines to send each line as separate event
+					lines := strings.Split(string(buf[:n]), "\n")
```

Comment:
Since you are reading 1024 bytes at a time, it is possible that this will break up long lines. Or reading could end in the middle of a line. Probably not a huge deal, but it would create weird line breaks for anything that is reading this.

I would suggest using a `bufio.Scanner` for reading instead because it will handle lines correctly and we won't have to worry about it here.

Reply:
Good call

Reply:
Fixed